src/py/crankshaft/crankshaft/clustering/kmeans.py

```diff
@@ -0,0 +1,17 @@
+from sklearn.cluster import KMeans
+import plpy
+
+def kmeans(query, no_clusters, no_init=20):
```

Comment:
So K-Means is a stochastic algorithm which is not necessarily guaranteed to find the right answer with just 1 run. The n_init variable is the number of times the algorithm runs after with it takes the best answer, the one with the best defined clusters. 

The number n_init depends a lot on the problem, the data space and the number of classes. It's not 100% possible to predict it ahead of time. 

The performance should scale linearly with n_init. The accuracy is almost impossible to predict. 

Happy to lower it for now. Was just airing on the side of getting good clusters